api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java

```diff
@@ -0,0 +1,90 @@
+import java.util.logging.Logger;
+
+@RestController
+public class CohortBuilderController implements CohortBuilderApiDelegate {
+
+    private static final Logger log = Logger.getLogger(CohortBuilderController.class.getName());
+
+    public static final String CRTIERIA_QUERY = "SELECT id, type, code, name, est_count, is_group, is_selectable, domain_id\n"
+            + "FROM `pmi-drc-api-test.synpuf.%s`\n"
+            + "WHERE parent_id = @parentId\n"
+            + "order by id asc";
+
+    @Override
+    public ResponseEntity<CriteriaListResponse> getCriteriaByTypeAndParentId(String type, String parentId) {
+
+        QueryResult result = getQueryResult(type, parentId);
+
+        Map<String, Integer> rm = getResultMapper(result);
+
+        CriteriaListResponse criteriaResponse = new CriteriaListResponse();
+        for (List<FieldValue> row : result.iterateAll()) {
+            final Criteria criteria = new Criteria();
+            criteria.setId(row.get(rm.get("id")).getLongValue());
+            criteria.setType(row.get(rm.get("type")).getStringValue());
+            criteria.setCode(row.get(rm.get("code")).getStringValue());
+            criteria.setName(row.get(rm.get("name")).getStringValue());
+            criteria.setCount(row.get(rm.get("est_count")).isNull() ? 0 : row.get(rm.get("est_count")).getLongValue());
+            criteria.setGroup(row.get(rm.get("is_group")).getBooleanValue());
+            criteria.setSelectable(row.get(rm.get("is_selectable")).getBooleanValue());
+            criteria.setDomainId(row.get(rm.get("domain_id")).isNull() ? null : row.get(rm.get("domain_id")).getStringValue());
+            criteriaResponse.addItemsItem(criteria);
+        }
+
+        return ResponseEntity.ok(criteriaResponse);
+    }
+
+    protected QueryResult getQueryResult(String type, String parentId) {
+        BigQuery bigquery =
+                new BigQueryOptions.DefaultBigqueryFactory().create(BigQueryOptions.getDefaultInstance());
+
+        QueryRequest queryRequest =
+                QueryRequest.newBuilder(getQueryString(type))
+                        .addNamedParameter("parentId", QueryParameterValue.int64(new Integer(parentId)))
+                        .setUseLegacySql(false)
+                        .build();
+
+        // Execute the query.
+        QueryResponse response = bigquery.query(queryRequest);
+
+        // Wait for the job to finish
+        while (!response.jobCompleted()) {
+            response = bigquery.getQueryResults(response.getJobId());
+        }
+
+        // Check for errors.
+        if (response.hasErrors()) {
+            String firstError = "";
+            if (response.getExecutionErrors().size() != 0) {
+                firstError = response.getExecutionErrors().get(0).getMessage();
+            }
+            throw new RuntimeException(firstError);
+        }
+
+        return response.getResult();
+    }
+
+    protected Map<String, Integer> getResultMapper(QueryResult result) {
```

Comment:
Eventually we'll probably want to move this and some of the code above into some BigQuery utils class. (Fine to leave it here for now.)